src/stanza.c

```diff
@@ -431,7 +431,7 @@ int xmpp_stanza_to_text(xmpp_stanza_t *stanza,
     ret = _render_stanza_recursive(stanza, buffer, length);
     if (ret < 0) return ret;
 
-    if (ret > length - 1) {
+    if ((unsigned)ret > length - 1) {
```

Comment:
done

AFAIK `size_t` is always unsigned, there's also `ssize_t` which would be obsolete on such a platform :)